### PR TITLE
feat: get employee for selection box

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] get list employees for selection box

#### What are the relevant Git tickets?
https://www.notion.so/dwarves/API-to-GET-employee-list-For-selection-box-f13e313855784c78b8357891a093e033

#### Screenshots (if appropriate)
image

#### Request
```
curl --location --request GET 'http://localhost:8200/api/v1/employees?workingStatus=probation&preload=false' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM' \
--data-raw ''
--data-raw ''
```

#### Response
```
{
    "page": 0,
    "size": 50,
    "sort": "",
    "total": 1,
    "data": [
        {
            "id": "ecea9d15-05ba-4a4e-9787-54210e3b98ce",
            "createdAt": "2022-11-02T09:50:55.320669Z",
            "updatedAt": "2022-11-02T09:50:55.320669Z",
            "fullName": "Nguyễn Hoàng Huy",
            "displayName": "Huy Nguyen",
            "teamEmail": "huy123@d.foundation",
            "personalEmail": "hoanghuy123@gmail.com",
            "avatar": "https://s3-ap-southeast-1.amazonaws.com/fortress-images/2830497479497502617.png",
            "phoneNumber": "0123456789",
            "address": "chung cư Sunview Town, Gò Dưa, Thủ Đức",
            "mbti": "Defender",
            "gender": "Male",
            "horoscope": "Virgo",
            "birthday": "1990-01-02T00:00:00Z",
            "discordID": "",
            "githubID": "",
            "notionID": "",
            "status": "probation",
            "joinedDate": null,
            "leftDate": null,
            "accountStatus": "active",
            "seniority": null,
            "chapter": null,
            "lineManager": null,
            "positions": [],
            "stacks": [],
            "roles": [],
            "projects": []
        }
    ]
}
```

### CODE REVIEW

File: pkg/routes/v1.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ func loadV1Routes(r *gin.Engine, h *handler.Handler, repo store.DBRepo, s *store
 
 	// employees
 	v1.GET("/employees", mw.WithAuth, h.Employee.List)
+	v1.GET("/employees/basic", mw.WithAuth, h.Employee.ListForSelectionBox)
```
Comment (nguyennh4522) [inline pkg/routes/v1.go:23]: conflict with `/employees/:id`

File: pkg/routes/v1_test.go [hunk 2]
```diff
@@ -26,6 +26,12 @@ func Test_loadV1Routes(t *testing.T) {
 				Handler: "github.com/dwarvesf/fortress-api/pkg/handler/employee.IHandler.Create-fm",
 			},
 		},
+		"/api/v1/employees/basic": {
+			"GET": {
+				Method:  "GET",
+				Handler: "github.com/dwarvesf/fortress-api/pkg/handler/employee.IHandler.ListForSelectionBox-fm",
+			},
+		},
```
Comment (namnhce) [inline pkg/routes/v1_test.go:34]: Delete this one

File: pkg/store/employee/interface.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 type IStore interface {
 	Search(db *gorm.DB, query SearchFilter, pagination model.Pagination) (employees []*model.Employee, total int64, err error)
 	One(db *gorm.DB, id string) (employee *model.Employee, err error)
+	All(db *gorm.DB) (employee []*model.Employee, err error)
```
Comment (namnhce) [inline pkg/store/employee/interface.go:12]: Delete this one

File: pkg/view/employee.go [hunk 4]
```diff
@@ -100,6 +107,24 @@ type UpdateGeneralEmployeeResponse struct {
...
+		rs = append(rs, ListForSelectionBoxData{
+			ID:       e.ID,
+			FullName: e.FullName,
+			Avatar:   e.Avatar,
+		})
+	}
+
+	return rs
+}
+
```
Comment (namnhce) [inline pkg/view/employee.go:127]: Delete this one

**Review by namnhce — CHANGES_REQUESTED** [review]

File: pkg/routes/v1.go (hunk 1 above)
Comment (datphamcode295) [reply pkg/routes/v1.go]: updated

File: pkg/routes/v1_test.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/routes/v1_test.go]: updated

File: pkg/store/employee/interface.go (hunk 3 above)
Comment (datphamcode295) [reply pkg/store/employee/interface.go]: updated

File: pkg/view/employee.go (hunk 4 above)
Comment (datphamcode295) [reply pkg/view/employee.go]: updated

**Review by namnhce — APPROVED** [review]

**Pull request merged**